subsys/dfu/dfu_target/src/dfu_target_modem_delta.c

```diff
@@ -101,6 +101,25 @@ int dfu_target_modem_delta_init(size_t file_size, dfu_target_callback_t cb)
 		return -EFBIG;
 	}
 
+	if (IS_ENABLED(CONFIG_DFU_TARGET_MODEM_ERASE_ON_INIT)) {
+		dfu_target_modem_delta_erase(0, false);
```

Comment:
Should we not look at the return value and possibly return that?

Reply:
yup, big oversight there as it should fail if the erase failed.